### ISSUE TITLE
Allow to specify nullable column ordering behaviour

### ISSUE THREAD

**Issue by Edvinas01** [JetBrains/Exposed#478]
When sorting a table in descending order while using PostgreSQL, `NULL` values appear on top. However, I would like `NULL` values to appear last. To solve this issue PostgreSQL has `NULLS LAST` and `NULLS FIRST` options:
```sql
SELECT *
FROM table_name
ORDER BY column_name DESC NULLS LAST;

SELECT *
FROM table_name
ORDER BY column_name ASC NULLS FIRST;
```

However when using Exposed, there doesn't seem a way to specify this as I'm only able to supply a `vararg` of `Pair<Column<*>, Boolean>` when ordering my queries:
```kotlin
val columns: List<Pair<Column<*>, Boolean>> = ...
val query: Query = ...

query
    // Can't specify weather nulls should appear last or first.
    .orderBy(*columns)
    .limit(...)
    .map(...)
```

**Comment by husnjak** [comment]
When will this enhancement be released?

**Comment by KennethWussmann** [comment]
I think it's also not possible to write an extension for that, at least I couldn't find a place for that.

When the `orderByExpressions` get appended to the `queryBuilder` [here](https://github.com/JetBrains/Exposed/blob/b0fd2bca679a7792fa15d5d4ad98821c7f6cee63/exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Query.kt#L209), there is no way to intercept to inject a `NULLS LAST`. Maybe when you don't use a `LIMIT` and `FOR UPDATE` you could append it after the actual building is done, but that's not working for me.

**Comment by KennethWussmann** [comment]
Okay, I found a way to archive `NULLS LAST` via an exposed expression.
```kotlin
class ColumnNullsLast(private val col: Column<*>) : Expression<String>() {
    override fun toQueryBuilder(queryBuilder: QueryBuilder) = queryBuilder {
        append(col, " IS NULL, ", col)
    }
}

fun Column<*>.nullsLast() = ColumnNullsLast(this)
```
Usage:
```
query.orderBy(User.age.nullsLast() to SortOrder.DESC)
```

**Comment by ybznek** [comment]
instead of `ColumnNullsLast` `org.jetbrains.exposed.sql.IsNullOp` could be used

**Comment by MEJIOMAH** [comment]
NULLS FIRST in postgres use index scan. This solution use NULL FIRST statement. 

enum class NullOrder {
    FIRST,
    LAST
}

fun Query.orderBy(vararg expressions: Pair<NullsExpression, SortOrder>): Query {
    return orderBy(
        *expressions.map { (nullExpression, sortOrder) ->
            Pair(OrderWithNull(nullExpression.expression, sortOrder, nullExpression.nullOrder), sortOrder)
        }.toTypedArray()
    )
}

fun Expression<*>.nullsFirst(): NullsExpression = NullsExpression(this, NullOrder.FIRST)
fun Expression<*>.nullsLast(): NullsExpression = NullsExpression(this, NullOrder.LAST)

class NullsExpression(val expression: Expression<*>, val nullOrder: NullOrder)

private class OrderWithNull(
    private val expression: Expression<*>,
    private val sortOrder: SortOrder,
    private val nullOrder: NullOrder
) : Expression<String>() {
    override fun toQueryBuilder(queryBuilder: QueryBuilder): Unit = queryBuilder {
        append(expression)
        append(" ")
        append(sortOrder.name)
        when (nullOrder) {
            NullOrder.FIRST -> append(" NULLS FIRST ")
            NullOrder.LAST -> append(" NULLS LAST ")
        }

        append(", TRUE ")
    }
}

**Comment by MEJIOMAH** [comment]
NULLS FIRST/LAST  use index in postgres. This solution for null first

```kotlin
enum class NullOrder {
    FIRST,
    LAST
}

fun Query.orderBy(vararg expressions: Pair<NullsExpression, SortOrder>): Query {
    return orderBy(
        *expressions.map { (nullExpression, sortOrder) ->
            Pair(OrderWithNull(nullExpression.expression, sortOrder, nullExpression.nullOrder), sortOrder)
        }.toTypedArray()
    )
}

fun Expression<*>.nullsFirst(): NullsExpression = NullsExpression(this, NullOrder.FIRST)
fun Expression<*>.nullsLast(): NullsExpression = NullsExpression(this, NullOrder.LAST)

class NullsExpression(val expression: Expression<*>, val nullOrder: NullOrder)

private class OrderWithNull(
    private val expression: Expression<*>,
    private val sortOrder: SortOrder,
    private val nullOrder: NullOrder
) : Expression<String>() {
    override fun toQueryBuilder(queryBuilder: QueryBuilder): Unit = queryBuilder {
        append(expression)
        append(" ")
        append(sortOrder.name)
        when (nullOrder) {
            NullOrder.FIRST -> append(" NULLS FIRST ")
            NullOrder.LAST -> append(" NULLS LAST ")
        }

        append(", TRUE ")
    }
}```

**Comment by LukaszSkowronek** [comment]
Hi, is there any progress on that?

Regarding last comment it produces  'order by' query part as  e.g 
`ORDER BY PRIORITY DESC NULLS LAST, TRUE DESC, VALUABLE DESC NULLS LAST, TRUE DESC` 
Which results as wrong query.

**Comment by erwinw** [comment]
This got implemented in #1344 and several follow-ups from @Tapac; no release has been cut yet, but I'm guessing that won't take too long. Thanks Andrey!
